src/cmake-tools.ts

```diff
@@ -920,9 +912,10 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
   /**
    * Implementation of `cmake.setVariant`
    */
-  async setVariant() {
-    const ret = await this._variantManager.selectVariant();
-    if (ret) {
+  async setVariant(name?: string) {
+    // ! for compatibility with return code style...
+    const ret = ! await this._variantManager.selectVariant(name);
+    if (!ret) {
       await this.configure();
     }
     return ret;
```

Comment:
This return now becomes correct because I inverted the meaning of ret :). SelectVariant returns true when succeeding, but succeeding code for cmake commands is 0.

Reply:
Resolving this comment as duplicate.